tests/test_detectAndMeasure.py

```diff
@@ -217,6 +265,261 @@ def _check_diaSource(self, refSources, diaSource, refIds=None,
+        subtractTask = subtractImages.AlardLuptonPreconvolveSubtractTask()
+        scienceKernel = science.psf.getKernel()
+
+        def _detection_wrapper(polarity=1):
+            transients, transientSources = makeTestImage(seed=transientSeed, psfSize=2.4,
+                                                         nSrc=10, fluxLevel=1000.,
+                                                         noiseLevel=noiseLevel, noiseSeed=8)
+            difference = science.clone()
+            difference.maskedImage -= matchedTemplate.maskedImage
+            if polarity > 0:
+                difference.maskedImage += transients.maskedImage
+            else:
+                difference.maskedImage -= transients.maskedImage
+            score = subtractTask._convolveExposure(difference, scienceKernel, subtractTask.convolutionControl)
+            output = detectionTask.run(science, matchedTemplate, difference, score)
+            refIds = []
+            for diaSource in output.diaSources:
+                self._check_diaSource(transientSources, diaSource, refIds=refIds, scale=polarity)
+        _detection_wrapper(polarity=1)
+        _detection_wrapper(polarity=-1)
+
+    def test_detect_dipoles(self):
+        """Run detection on a difference image containing dipoles.
+        """
+        noiseLevel = 1.
+        staticSeed = 1
+        fluxLevel = 1000
+        fluxRange = 1.5
+        nSources = 10
+        offset = 1
+        dipoleFlag = "ip_diffim_DipoleFit_flag_classification"
+        kwargs = {"seed": staticSeed, "psfSize": 2.4, "fluxLevel": fluxLevel, "fluxRange": fluxRange,
+                  "nSrc": nSources}
+        science, sources = makeTestImage(noiseLevel=noiseLevel, noiseSeed=6, **kwargs)
+        matchedTemplate, _ = makeTestImage(noiseLevel=noiseLevel/4, noiseSeed=7, **kwargs)
+        difference = science.clone()
+        matchedTemplate.image.array[...] = np.roll(matchedTemplate.image.array[...], offset, axis=0)
+        matchedTemplate.variance.array[...] = np.roll(matchedTemplate.variance.array[...], offset, axis=0)
+        matchedTemplate.mask.array[...] = np.roll(matchedTemplate.mask.array[...], offset, axis=0)
+        difference.maskedImage -= matchedTemplate.maskedImage[science.getBBox()]
+        subtractTask = subtractImages.AlardLuptonPreconvolveSubtractTask()
+        scienceKernel = science.psf.getKernel()
+        score = subtractTask._convolveExposure(difference, scienceKernel, subtractTask.convolutionControl)
+        config = detectAndMeasure.DetectAndMeasureScoreTask.ConfigClass()
+        config.doApCorr = False
+        config.doMerge = False
+        config.doSkySources = False
+        config.doForcedMeasurement = False
+        detectionTask = detectAndMeasure.DetectAndMeasureScoreTask(config=config)
+        output = detectionTask.run(science, matchedTemplate, difference, score)
+        self.assertIn(dipoleFlag, output.diaSources.schema.getNames())
```

Comment:
Shouldn't that flag always be in the output schema, whether there are detected dipoles in the image or not?